src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -150,71 +188,139 @@ def validate_public_ip_type(namespace): # pylint: disable=unused-argument
     servers = []
     for item in namespace.servers if namespace.servers else []:
         try:
             socket.inet_aton(item) #pylint:disable=no-member
-            servers.append({'IpAddress': item})
+            servers.append(ApplicationGatewayBackendAddress(ip_address=item))
         except socket.error: #pylint:disable=no-member
-            servers.append({'Fqdn': item})
+            servers.append(ApplicationGatewayBackendAddress(fqdn=item))
     namespace.servers = servers
 
-def validate_cert(namespace):
-
-    params = [namespace.cert_data, namespace.cert_password]
-    if all([not x for x in params]):
-        # no cert supplied -- use HTTP
-        namespace.http_listener_protocol = 'http'
-        if not namespace.frontend_port:
-            namespace.frontend_port = 80
-    else:
-        # cert supplied -- use HTTPS
-        if not all(params):
-            raise argparse.ArgumentError(
-                None, 'To use SSL certificate, you must specify both the filename and password')
+def validate_subnet_name_or_id(namespace):
+    """ Validates a subnet ID or, if a name is provided, formats it as an ID. """
+    if namespace.virtual_network_name is None and namespace.subnet is None:
+        return
+    if namespace.subnet == '':
+        return
+    # error if vnet-name is provided without subnet
+    if namespace.virtual_network_name and not namespace.subnet:
+        raise CLIError('You must specify --subnet name when using --vnet-name.')
 
-        # extract the certificate data from the provided file
-        with open(namespace.cert_data, 'rb') as f:
-            contents = f.read()
-            base64_data = base64.b64encode(contents)
-            try:
-                namespace.cert_data = base64_data.decode('utf-8')
-            except UnicodeDecodeError:
-                namespace.cert_data = str(base64_data)
+    # determine if subnet is name or ID
+    is_id = is_valid_resource_id(namespace.subnet)
 
-        # change default to frontend port 443 for https
-        namespace.http_listener_protocol = 'https'
-        if not namespace.frontend_port:
-            namespace.frontend_port = 443
+    # error if vnet-name is provided along with a subnet ID
+    if is_id and namespace.virtual_network_name:
+        raise argparse.ArgumentError(None, 'Please omit --vnet-name when specifying a subnet ID')
+    elif not is_id and not namespace.virtual_network_name:
+        raise argparse.ArgumentError(None,
+                                     'Please specify --vnet-name when specifying a subnet name')
+    if not is_id:
+        namespace.subnet = resource_id(
+            subscription=get_subscription_id(),
+            resource_group=namespace.resource_group_name,
+            namespace='Microsoft.Network',
+            type='virtualNetworks',
+            name=namespace.virtual_network_name,
+            child_type='subnets',
+            child_name=namespace.subnet)
 
 # COMMAND NAMESPACE VALIDATORS
 
```

Comment:
Which code? Github would suggest the code above, but I would assume you mean the code below.

---
In reply to: [73620167](https://github.com/Azure/azure-cli/pull/616#discussion_r73620167) [](ancestors = 73620167)